File: trunk/proj/CoreLibrary/src/com/jasonzqshen/familyaccounting/core/exception/TransactionDataFileFormatException.java
package com.jasonzqshen.familyaccounting.core.exception;

public class TransactionDataFileFormatException extends Exception {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	public final String FILE_PATH;

	public TransactionDataFileFormatException(String filePath) {
		super(String.format("Transaction data file %s contains format error",
				filePath));
		FILE_PATH = filePath;
	}
}

File: proj/WinStore_FamilyAccounting/WinStore_FamilyAccountingCore/Exceptions/FormatExceptions/TransactionDataFileFormatException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WinStore_FamilyAccountingCore.Exceptions.FormatExceptions
{
    public class TransactionDataFileFormatException : Exception
    {
        public String FilePath { get; protected set; }

        public TransactionDataFileFormatException(String filePath)
            : base(String.Format("Transaction data file {0} contains format error",
                    filePath))
        {
            FilePath = filePath;
        }
    }
}
